src/query_builder.rs

```diff
@@ -0,0 +1,233 @@
+            {wit_source}
+            ",
+            name = self.shared.name,
+            force = if self.shared.force { "OR REPLACE" } else { "" },
+            abi = self.shared.abi_type,
+        )
+    }
+}
+
+impl QueryBuilder for TableFunction {
+    fn build(&self, args: &mut Vec<String>) -> String {
+        let wasm_source = self.shared.wasm_source.build(args);
+        let wit_source = self
+            .shared
+            .wit_source
+            .as_ref()
+            .map(|w| format!("WITH WIT {}", w.build(args)))
+            .unwrap_or_default();
+        format!(
+            "
+            CREATE {force} FUNCTION {name}
+            RETURNS TABLE
+            AS WASM ABI {abi} 
+            {wasm_source}
+            {wit_source}
+            ",
+            name = self.shared.name,
+            force = if self.shared.force { "OR REPLACE" } else { "" },
+            abi = self.shared.abi_type,
+        )
+    }
+}
+
+impl QueryBuilder for AggregateFunction {
+    fn build(&self, args: &mut Vec<String>) -> String {
+        let imports = match self.aggregate_imports.as_ref() {
+            Some(imports) => imports.clone(),
+            None => {
+                if let Some(wit_source) = &self.shared.wit_source {
+                    // Try auto-detect from wit
+                    let functions = try_extract_member_functions_from_wit(wit_source.unpack());
+
+                    ManualAggregateImports {
+                        init: functions.get(&MemberFunction::Init).cloned(),
+                        iter: functions.get(&MemberFunction::Iterate).cloned(),
+                        merge: functions.get(&MemberFunction::Merge).cloned(),
+                        terminate: functions.get(&MemberFunction::Terminate).cloned(),
+                        serialize: functions.get(&MemberFunction::Serialize).cloned(),
+                        deserialize: functions.get(&MemberFunction::Deserialize).cloned(),
+                    }
+                } else {
+                    eprintln!("Cannot auto-detect functions without WIT source. Please specify a .wit file with --wit or manually provide the required functions.");
+                    exit(1);
+                }
+            }
+        };
+
+        // Validate the imports
+        imports.exit_if_not_valid();
+
+        let wasm_source = self.shared.wasm_source.build(args);
+        let wit_source = self
+            .shared
+            .wit_source
+            .as_ref()
+            .map(|w| format!("WITH WIT {}", w.build(args)))
+            .unwrap_or_default();
+        let args = if self.args.len() == 1 {
+            // The name of the argument is omitted for single args
+            self.args[0].clone()
+        } else {
+            // A unique argument name is generated for each args, e.g.
+            // "a int, b int, c int, ..."
+            let generate_var_names = |(i, arg_type)| {
+                let var_name_as_letter = ('a' as usize + i) as u8 as char;
```

Comment:
So, 26 args is the max?  Kind of an edge case, but maybe we should emit an error if there are more so that we don't generate bad SQL.

Reply:
Done, returning an error message.